core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -628,9 +628,9 @@ class KafkaController(val config: KafkaConfig,
       topicDeletionManager.failReplicaDeletion(newOfflineReplicasForDeletion)
     }
 
-    // If replica failure did not require leader re-election, inform brokers of the offline brokers
+    // If no partitions have ISR change, inform brokers of the offline brokers
```

Comment:
How about changing the comments to  the following?

If no partition has changed leader or ISR, no UpdateMetadataReuqest is sent through PartitionStateMachine and ReplicaStateMachine. In that case, we want to send an UpdateMetadataReuqest explicitly to propagate the information about the new offline brokers.